### PR TITLE
 161445502 Create authentication endpoints

### PR DESCRIPTION
**Pull request by charitymarani**
**Introduction**
This PR develops authentication endpoints register login and logout

**Description of Task to be completed:**
On sending requests to the endpoints appropriate response should be received

**How to manually test**

`$ pip install virtualenv`
NB This install virtualenv Globally(virtual environment allows you to create independent envs isolated from your actual physical machine)
`$ virtualenv -p python3 myenv`
`$ source myenv/bin/activate`
Clone this repository: `$ git clone https://github.com/charitymarani/store_manager_api_v2.git`
navigate into the repository after cloning
`$ git checkout ft-authentication-endpoints-161445502 to switch from master to this branch`
`$ pip install -r requirements.txt`
Launch postman

`POST /api/v2/regiter`
`POST /api/v2/login`
`POST /api/v2/logout`
`GET /api/v2/users`
`GET /api/v2/users/username`

`headers = {content_type:application/json}`
For register and logout,grab the jwt token from the login response and add authorization header:`Bearer + <jwt>`

**Any background information**
Successful request returns JSON formatted response appropriately
**Relevant story**
[161445502](https://www.pivotaltracker.com/story/show/161445502)

### CODE REVIEW

File: application/__init__.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+    @jwt.user_claims_loader
+    def add_claims_to_access_token(user_object):
+        '''add role claims to access token'''
+        return {'role': user_object['role']}
+
+    @jwt.user_identity_loader
+    def user_identity_lookup(user_object):
+        '''set token identity from user_object passed to username'''
+        print(user_object)
```
Comment (Bonifase) [inline application/__init__.py]: Why is this line of code here?

File: application/utils.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
+from manage import DbSetup
+
+conn = DbSetup("development")
+cur = conn.cursor()
+
+
+def list_iterator(list):
+    for i in list:
```
Comment (Bonifase) [inline application/utils.py]: avoid using single letter variables like i, e.t.c. Kindly use more descriptive variable names

File: manage.py [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+        return cur
+
+    def commit(self):
+        '''commits changes to db'''
+        conn = psycopg2.connect(self.connection_string)
+        conn.commit()
+
+    def tables(self):
+
+        q1 = """CREATE TABLE IF NOT EXISTS products (
```
Comment (Bonifase) [inline manage.py]: Use more descriptive variable names, avoid names like q1

**Review by Bonifase — COMMENTED** [review]
Good work, look into the comments I have left

File: application/__init__.py (hunk 1 above)
Comment (charitymarani) [reply application/__init__.py]: The print statement or the user_identity_lookup() function?
The print statement was for debugging, lemme remove it

**Pull request merged**